packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.tsx

```diff
@@ -0,0 +1,472 @@
+import React, {
+  useState,
+  useCallback,
+  useMemo,
+  useEffect,
+  useContext,
+} from 'react';
+import {
+  Stripe,
+  StripeCardElement,
+  PaymentMethod,
+  StripeError,
+} from '@stripe/stripe-js';
+import { Plan, Profile, Customer } from '../../../store/types';
+import { PromiseResolved } from '../../../lib/types';
+import { State as ValidatorState } from '../../../lib/validator';
+
+import { useAwait, PromiseState } from 'fxa-react/lib/hooks';
+import { useNonce } from '../../../lib/hooks';
+import { getErrorMessage } from '../../../lib/errors';
+
+import PlanDetails from '../../../components/PlanDetails';
+import Header from '../../../components/Header';
+import PaymentForm, {
+  PaymentFormProps,
+} from '../../../components/PaymentFormV2';
+import ErrorMessage from '../../../components/ErrorMessage';
+import AcceptedCards from '../AcceptedCards';
+
+import * as Amplitude from '../../../lib/amplitude';
+import { Localized } from '@fluent/react';
+import * as apiClient from '../../../lib/apiClient';
+
+import './index.scss';
+
+type PaymentError = { type: string; error: boolean };
+
+type RetryStatus = null | {
+  invoiceId: string;
+  paymentIntentStatus: string;
+};
+
+export type SubscriptionCreateProps = {
+  isMobile: boolean;
+  profile: Profile;
+  customer: Customer | null;
+  selectedPlan: Plan;
+  refreshSubscriptions: Function;
+  validatorInitialState?: ValidatorState;
+  createCustomerInitialState?: PromiseState<
+    PromiseResolved<ReturnType<typeof apiClient.apiCreateCustomer>>
+  >;
+  createSubscriptionInitialState?: PromiseState<
+    PromiseResolved<
+      ReturnType<typeof apiClient.apiCreateSubscriptionWithPaymentMethod>
+    >
+  >;
+  retryInvoiceInitialState?: PromiseState<
+    PromiseResolved<ReturnType<typeof apiClient.apiRetryInvoice>>
+  >;
+  apiClientOverrides?: Partial<typeof apiClient>;
+};
+
+export const SubscriptionCreate = ({
+  isMobile = false,
+  profile,
+  customer,
+  selectedPlan,
+  refreshSubscriptions,
+  validatorInitialState,
+  createCustomerInitialState,
+  createSubscriptionInitialState,
+  retryInvoiceInitialState,
+  apiClientOverrides = {},
+}: SubscriptionCreateProps) => {
+  const {
```

Comment:
A big departure from the previous payment page: Dropping use of Redux in favor of our new `useAwait` hook and managing API usage here locally.